scaladoc/src/dotty/tools/scaladoc/site/SidebarParser.scala

```diff
@@ -30,16 +32,43 @@ object Sidebar:
 
   private object RawInputTypeRef extends TypeReference[RawInput]
 
-  private def toSidebar(r: RawInput)(using CompilerContext): Sidebar = r match
+  private def pageWithNoTitle(content: String | java.io.File): String =
+    val fileContent = content match {
+    case file: java.io.File => Source.fromFile(file).getLines().mkString("\n")
+    case str: String => str
+    }
+    val lines = fileContent.split("\n")
+    lines.zipWithIndex
+    .find { case (line, i) => line.trim.startsWith("page:") && !lines(i - 1).contains("- title:") }
+    .map(_._1.trim.stripPrefix("page:"))
+    .getOrElse("")
+
+  private def toSidebar(r: RawInput, content: String | java.io.File)(using CompilerContext): Sidebar = r match
     case RawInput(title, page, index, subsection, dir, hidden) if page.nonEmpty && index.isEmpty && subsection.isEmpty() =>
+      val sidebarPath = content match
+        case s: String => Paths.get(s)
+        case f: java.io.File => f.toPath()
+      val basePath = sidebarPath.getParent().resolve("_docs")
+      val pagePath = basePath.resolve(page)
+      if !Files.exists(pagePath) then
+        report.error(s"Page $page does not exist.")
       Sidebar.Page(Option.when(title.nonEmpty)(title), page, hidden)
     case RawInput(title, page, index, subsection, dir, hidden) if page.isEmpty && (!subsection.isEmpty() || !index.isEmpty()) =>
-      Sidebar.Category(Option.when(title.nonEmpty)(title), Option.when(index.nonEmpty)(index), subsection.asScala.map(toSidebar).toList, Option.when(dir.nonEmpty)(dir))
+      Sidebar.Category(Option.when(title.nonEmpty)(title), Option.when(index.nonEmpty)(index), subsection.asScala.map(toSidebar(_, content)).toList, Option.when(dir.nonEmpty)(dir))
     case RawInput(title, page, index, subsection, dir, hidden) =>
-      report.error(s"Error parsing YAML configuration file.\n$schemaMessage")
+      if title.isEmpty() && index.isEmpty() then
+        val page = pageWithNoTitle(content).trim()
```

Comment:
Actually I don’t understand this part. Why do we call `pageWithNoTitle` instead of using the `page` field of the scrutinee `r`?

Reply:
Because in some cases (including the case where a title is missing), the Try(ReadValue) function in load() does not succeed but still passes into the box where I perform my validations. So r = RawInput(,,,[],,false) and I can't get anything out of it.

Reply:
OK I see. I think in that case we should not try to manually re-parse the sidebar to find out the `page`. Ultimately, we should probably use a different approach for parsing (maybe with [scala-yaml](https://virtuslab.github.io/scala-yaml/)?) that would allow us to navigate in the YAML data structure, but that’s a different topic.

I suggest we provide the best error messages we can without re-parsing the YAML in such a *ad-hoc* way. In that case, if we can’t get the `page`, just report “`title` is not provided for some page”. I know this is not very satisfactory, but I think we should really not try to re-parse the YAML text…

Reply:
Indeed it would be much better, I am working on it, thank you very much!